src/pk/asn1/der/utf8/der_decode_utf8_string.c

```diff
@@ -53,21 +53,23 @@ int der_decode_utf8_string(const unsigned char *in,  unsigned long inlen,
 
    /* proceed to decode */
    for (y = 0; x < inlen; ) {
-      /* get first byte */
+      /* read first byte */
       tmp = in[x++];
 
-      /* count number of bytes */
+      /* count number of left-shifts needed to get zero in most-sig bit */
       for (z = 0; (tmp & 0x80) && (z <= 4); z++, tmp = (tmp << 1) & 0xFF);
 
+      /* for valid utf8, z is in {0,2,3,4}.
+         if z>0, then z-1 equals the number of additional bytes to read */
```

Comment:
I'd move that second line below to the `--z` operation.

Reply:
I added the second comment line to make the length check `x+(z-1) > inlen` more comprehensible.  If I move that comment below, then there is no longer a hint to explain `x+(z-1) > inlen`.

Maybe the thing to do here is move `x+(z-1) > inlen` into a separate if-statement.